File: data_structure/lab_pracatice/binary_search.c
#include<stdio.h>
 void main(){
    int arr[10]={11,22,33,44,55,66,77,88,99,100};
    int mid;
    int l=0,h=10;
    int is =0;
    int key =100;
    while(l<h){
        mid = (l+h)/2;
        if(arr[mid]==key){
           is =1;
        }
         if(arr[mid]>key){
            h=mid-1;
        }
        else{
            l=mid+1;
        }
    }
        if(is==1)
    printf("ok");
    else
    printf("no");
 }

File: java/binary_search.java
import java.lang.*;
import java.util.Scanner;
class binary_search{
    
    public static void main(String args []){
     Scanner sc = new  Scanner(System.in);
     int n = sc.nextInt();
     int arr[] = new int[n];
     for(int i=0;i<n;i++){
        arr[i] = sc.nextInt();
     }
     int key = sc.nextInt();
    int isFound =0;
     int mid,i=0;
     while(i<n){
        mid = (i+n)/2;
        if(arr[i]==key)
     {
        isFound =1;
        break;
     }
     if(key < arr[mid]){
        n = mid;
     }else{
        i = mid+1;
     }
     }
     if(isFound==1)
     System.out.println("found");
     else
     System.out.println("not");
     
    }
}